### DISCUSSION TITLE
Compute an overall metric of multi-label segmentation problem

### DISCUSSION THREAD

**Discussion by Yangtting** [Project-MONAI/tutorials#1245] [Q&A]
Hi!  I'm following the code _unetr_btcv_segmentation_3d.ipynb_ for my multi-label (two targets and background) segmentation task. And I successfully complete the training and the validation and the test processes. In the test process, I checked the metrics of each label of each sample in the test set. However, I want to merge these label to compute the overall metric of the sample, could you please help me, thanks a lot.

```
import numpy as np
def dice(x, y):
   intersect = np.sum(np.sum(np.sum(x * y)))
   y_sum = np.sum(np.sum(np.sum(y)))
   if y_sum == 0:
       return 0.0
   x_sum = np.sum(np.sum(np.sum(x)))
   return 2 * intersect / (x_sum + y_sum)
os.chdir(r"C:\Auto segmentation\cal_ar\unetr") 
from metrics import recall, precision, specificity, jac
from medpy import metric

post_pred = Compose([EnsureType(), AsDiscrete(argmax=True, to_onehot=3)])
post_label = Compose([EnsureType(), AsDiscrete(to_onehot=3)])
model.load_state_dict(torch.load(
    os.path.join(save_path, "best_metric_model.pth")))
model.eval()
with torch.no_grad():
    i=1
    test_dice_list_case1 = []
    test_recall_list_case1 = []
    test_HD_list_case1 = []
    for i, batch in enumerate(test_org_loader):
        val_inputs, val_labels = (batch["image"].cuda(), batch["label"].cuda())
        img_name = batch["image_meta_dict"]["filename_or_obj"][0].split("/")[-1]
        print("Inference on case {}".format(img_name))
        val_outputs = sliding_window_inference(val_inputs, (128,128,96), 4, model, overlap=0.8)
        val_outputs = torch.softmax(val_outputs, 1).cpu().numpy()
        val_outputs = np.argmax(val_outputs, axis=1).astype(np.uint8)
        val_labels = val_labels.cpu().numpy()[:, 0, :, :, :]
        test_dice_list_sub = []
        test_recall_list_sub = []
        test_HD_list_sub = []
        for i in range(1, 3):
            organ_Dice = dice(val_outputs[0] == i, val_labels[0] == i)
            test_dice_list_sub.append(organ_Dice)
           
            organ_recall = recall(val_outputs[0] == i, val_labels[0] == i)
            test_recall_list_sub.append(organ_recall)
           
            organ_HD = metric.binary.hd95(val_outputs[0] == i, val_labels[0] == i)
            test_HD_list_sub.append(organ_HD)
           
        test_dice_list_case1.append(test_dice_list_sub)
        test_recall_list_case1.append(test_recall_list_sub)
        test_HD_list_case1.append(test_HD_list_sub)
```

**Comment by KumoLiu** [comment]
Hi @Yangtting, you could directly calculate these metrics based on your output and targets.
https://github.com/Project-MONAI/MONAI/blob/b85e2c6751d525488dddc2e3b0b05576d81208be/monai/metrics/meandice.py#L22
https://github.com/Project-MONAI/MONAI/blob/b85e2c6751d525488dddc2e3b0b05576d81208be/monai/metrics/confusion_matrix.py#L25
Hope it can help you, thanks!
